tests/unit/test_adapter.py

```diff
@@ -116,35 +117,55 @@ def test_database_and_catalog_settings(self):
             )
 
     def test_databricks_sql_connector_connection_lt_2_0(self):
-        def connect(
-            server_hostname, http_path, access_token, session_configuration, _user_agent_entry
-        ):
-            self.assertEqual(server_hostname, "yourorg.databricks.com")
-            self.assertEqual(http_path, "sql/protocolv1/o/1234567890123456/1234-567890-test123")
-            self.assertEqual(access_token, "dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
-            self.assertEqual(session_configuration["spark.sql.ansi.enabled"], "true")
-            self.assertNotIn(CATALOG_KEY_IN_SESSION_PROPERTIES, session_configuration)
-            self.assertEqual(_user_agent_entry, f"dbt-databricks/{__version__.version}")
+        def connect_func(dbt_invocation_env):
+            def connect(
+                server_hostname, http_path, access_token, session_configuration, _user_agent_entry
+            ):
+                self.assertEqual(server_hostname, "yourorg.databricks.com")
+                self.assertEqual(http_path, "sql/protocolv1/o/1234567890123456/1234-567890-test123")
+                self.assertEqual(access_token, "dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
+                self.assertEqual(session_configuration["spark.sql.ansi.enabled"], "true")
+                self.assertNotIn(CATALOG_KEY_IN_SESSION_PROPERTIES, session_configuration)
+                self.assertEqual(
+                    _user_agent_entry,
+                    f"dbt-databricks/{__version__.version} ({dbt_invocation_env})",
+                )
+
+            return connect
 
-        self._test_databricks_sql_connector_connection("1.0.2", connect)
+        self._test_databricks_sql_connector_connection("1.0.2", connect_func("manual"))
+        with envvar(DBT_INVOCATION_ENV, "dbt_invocation_env"):
```

Comment:
Can we use `mock.patch.dict` to patch `os.environ` here?

Reply:
Could you show me how to use it? Thanks.